blocks/DataOutput/DataOutput.js

```diff
@@ -1,61 +1,88 @@
+
+/** @typedef {import('../../submodules/upload-client/upload-client.js').UploadcareFile[]} FileList} */
 
 export class DataOutput extends Block {
+  processInnerHtml = true;
+
+  init$ = {
+    output: null,
+    filesData: null,
+  };
+
+  get dict() {
+    return DataOutput.dict;
+  }
+
   initCallback() {
-    let from = this.getAttribute('from');
-    this.sub(
-      from || DataOutput.defaultFrom,
-      (/** @type {import('../../submodules/upload-client/upload-client.js').UploadcareFile[]} */ data) => {
-        if (!data) {
-          this.innerHTML = '';
-          return;
-        }
-        if (this.hasAttribute(DataOutput.fireEventAttrName)) {
-          this.dispatchEvent(
-            new CustomEvent(DataOutput.outputEventName, {
-              bubbles: true,
-              composed: true,
-              detail: {
-                timestamp: Date.now(),
-                ctxName: this.ctxName,
-                data,
-              },
-            })
-          );
-        }
-        if (this.hasAttribute(DataOutput.templateAttrName)) {
-          let tpl = this.getAttribute(DataOutput.templateAttrName);
-          let html = '';
-          data.forEach((fileItem) => {
-            let itemHtml = tpl;
-            for (let prop in fileItem) {
-              itemHtml = itemHtml.split(`{{${prop}}}`).join(fileItem[prop]);
-            }
-            html += itemHtml;
-          });
-          this.innerHTML = html;
-        }
-        this.value = data;
-        if (this.hasAttribute(DataOutput.formValueAttrName)) {
-          if (!this._input) {
-            /** @private */
-            this._input = document.createElement('input');
-            this._input.type = 'text';
-            this.appendChild(this._input);
-          }
-          this._input.value = JSON.stringify(data);
-        }
-        if (this.hasAttribute(DataOutput.consoleAttrName)) {
-          console.log(data);
+    this.sub('output', (data) => {
+      if (!data) {
+        return;
+      }
+      this.value = data;
+
+      if (this.hasAttribute(this.dict.FIRE_EVENT_ATTR)) {
+        this.dispatchEvent(
+          new CustomEvent(this.dict.EVENT_NAME, {
+            bubbles: true,
+            composed: true,
+            detail: {
+              timestamp: Date.now(),
+              ctxName: this.ctxName,
+              data,
+            },
+          })
+        );
+      }
+
+      if (this.hasAttribute(this.dict.FORM_VALUE_ATTR)) {
+        if (!this._input) {
+          /** @private */
+          this._input = document.createElement('input');
+          this._input.type = 'text';
+          this._input.hidden = true;
+          this.appendChild(this._input);
         }
+        this._input.value = JSON.stringify(data);
+      }
+
+      if (this.hasAttribute(this.dict.CONSOLE_ATTR)) {
+        console.log(data);
+      }
+    });
+
+    this.sub(this.dict.SRC_CTX_KEY, async (/** @type {FileList} */ data) => {
```

Comment:
There we have an extended `UploadcareFile` type. Probably it's better to create a separate extended type for it. Could you add a TODO comment? 

https://github.com/uploadcare/uc-blocks/blob/dev/abstract/Block.js#L519

Reply:
done